Copy the wasm-wasi-component module into build/lib/unit/modules/ as part of the build process

ac000 opened the pull request:
The first patch is just a clean up patch that fixes some indentation.

The second patch copies the wasm-wasi-component .so into `build/lib/unit/modules/` with the right name. This matches the behaviour of the other language modules and is needed by things like the pytests.

ac000 commented:
- Use install(1) instead of cp(1)
- Commit message tweaks

```
$ git range-diff a67b6412...4a2a43fd
1:  a67b6412 ! 1:  4a2a43fd auto, wasm-wc: Copy the .so into build/lib/unit/modules/
    @@ Metadata
      ## Commit message ##
         auto, wasm-wc: Copy the .so into build/lib/unit/modules/
     
    -    Normally when the language modules are built, they are copied into the
    -    build/lib/unit/modules/ directory (as part of the build, not install
    -    step).
    +    Normally when the language modules are built, they are built directly
    +    into the build/lib/unit/modules/ directory.
     
         This then allows Unit to find them without being installed. This is
         useful for things like the pytests.
     
    -    This wasn't happening for the wasm-wasi-componment language mpdule. So
    -    we now copy it over and give it the right name as part of the make/build
    +    This wasn't happening for the wasm-wasi-component language module. So we
    +    now copy it over and give it the right name as part of the make/build
         process.
     
         Reported-by: Andrei Zeliankou <zelenkov@nginx.com>
    +    Fixes: 4e6d7e876 ("Wasm-wc: Wire it up to the build system")
         Signed-off-by: Andrew Clayton <a.clayton@nginx.com>
     
      ## auto/modules/wasm-wasi-component ##
    @@ auto/modules/wasm-wasi-component: ${NXT_WCM_MODULE}:     ${NXT_WCM_MOD_CARGO}
      
      ${NXT_WCM_MOD_CARGO}: ${NXT_WCM_DEPS}
        $NXT_CARGO_CMD
    -+  @cp $NXT_WCM_MOD_CARGO build/lib/unit/modules/${NXT_WCM_MOD_NAME}
    ++  @install -p $NXT_WCM_MOD_CARGO \\
    ++          build/lib/unit/modules/${NXT_WCM_MOD_NAME}
      
      install: ${NXT_WCM_MODULE}-install
      
```

callahad reviewed (approved):
Seems reasonable, two nits/questions:

- Do we need to worry about updating a `clean` target anywhere, or including this in the `-uninstall` target?
- Do we need a `mkdir -p build/lib/unit/modules/` before calling `install`?

ac000 commented:
> Seems reasonable, two nits/questions:
> 
>     * Do we need to worry about updating a `clean` target anywhere, or including this in the `-uninstall` target?

No. This is copied as part of the build process (make not make install). Other language modules build the library directly into the directory. E.g

```
$NXT_BUILD_DIR/lib/unit/modules/${NXT_PYTHON_MODULE}.unit.so:   $nxt_objs
        \$(PP_LD) \$@
        \$(v)\$(NXT_MODULE_LINK) -o \$@ \\
        $nxt_objs $NXT_PYTHON_LIBS $NXT_PYTHON_LDFLAGS $NXT_LD_OPT
```

This is just the Unit internal build directory.

>     * Do we need a `mkdir -p build/lib/unit/modules/` before calling `install`?

It's already created by `./configure`

ac000 commented:
Rebased with master

```
$ git range-diff 4a2a43fd...b5fe3eaf
 -:  -------- >  1:  d62a5e2c contrib: updated njs to 0.8.5
 -:  -------- >  2:  a9aa9e76 python: Support application factories
 -:  -------- >  3:  d67d3501 tests: Add tests for python application factories
 -:  -------- >  4:  36213522 Fix certificate deletion for array type certificates
 -:  -------- >  5:  ff6d5045 python: Constify some local static variables
 -:  -------- >  6:  0bd18ecd Packages: clean up EOL debian-based distributions
 -:  -------- >  7:  151305eb Packages: added Ubuntu 24.04 "noble" support
 -:  -------- >  8:  6c04c7dc Packages: don't redefine FORTIFY_SOURCE on Ubuntu
 -:  -------- >  9:  804a12fe Packages: remove support for EOL Fedora versions (35-38)
 -:  -------- > 10:  929a275a Packages: removed CentOS 6 leftover
 -:  -------- > 11:  63fb95ed Packages: removed CentOS 7 due to EOL
 1:  b1cb7a0b = 12:  0e6cc6dc auto: Fix some indentation in auto/modules/wasm-wasi-component
 2:  4a2a43fd = 13:  b5fe3eaf auto, wasm-wc: Copy the .so into build/lib/unit/modules/
```
